Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py

```diff
@@ -100,6 +97,104 @@ def _oproxy_authorize(self) -> Tuple[str, str, str, str, int]:
+        - After 48 hours check for access token once every 60 minutes max.
+        Successful fetch of access token resets the counters.
+
+        Returns: The oproxy response or raising a DemistoException
+
+        """
+        err_msg = 'We have found out that your recent attempts to authenticate against the CDL server have failed. ' \
+                  'Therefore we have limited the number of calls that the CDL integration performs. ' \
+                  'If you wish to try authenticating again, please run the `cdl-reset-authentication-timeout` ' \
+                  'command and retry.'
+        integration_context = demisto.getIntegrationContext()
+        first_failure_time = integration_context.get(FIRST_FAILURE_TIME_CONST)
+        last_failure_time = integration_context.get(LAST_FAILURE_TIME_CONST)
+
+        if first_failure_time and last_failure_time:
+            first_failure_datetime = datetime.fromtimestamp(first_failure_time)
+            last_failure_datetime = datetime.fromtimestamp(last_failure_time)
+            now_datetime = datetime.now()
+            time_from_first_failure = now_datetime - first_failure_datetime
+            time_from_last_failure = now_datetime - last_failure_datetime
+
+            if time_from_first_failure < timedelta(hours=1):
+                if time_from_last_failure < timedelta(minutes=1):
+                    raise DemistoException(err_msg)
+            elif time_from_first_failure < timedelta(hours=48):
+                if time_from_last_failure < timedelta(minutes=10):
+                    raise DemistoException(err_msg)
+            else:
+                if time_from_last_failure < timedelta(minutes=60):
+                    raise DemistoException(err_msg)
+
+        return self._get_access_token()
+
+    def _get_access_token(self) -> dict:
+        """ Performs an http request to oproxy-cdl access token endpoint
+        In case of failure, handles the error, otherwise reset the failure counters and return the response
+
+        Returns: The oproxy response or raising a DemistoException
+
+        """
+        try:
+            oproxy_response = self._http_request('POST',
+                                                 '/cdl-token',
+                                                 json_data={'token': get_encrypted(self.refresh_token, self.enc_key)},
+                                                 timeout=(60 * 3, 60 * 3),
+                                                 retries=3,
+                                                 backoff_factor=10,
+                                                 status_list_to_retry=[400])
+        except DemistoException as e:
+            if self._is_bad_request_error(e):
+                # if the error's status code is 400, update the failure time counters
+                integration_context = demisto.getIntegrationContext()
+                current_time = datetime.now().timestamp()
```

Comment:
```suggestion
                current_time = datetime.now(timezone.utc).isoformat()
```
Save something readable to the context